### ISSUE TITLE
[6.x] AddOns

### ISSUE THREAD

**Issue by hoxton22** [TrinityCore/TrinityCore#14949]
What steps will reproduce the problem?
Download Total RP 3 and connect two players.

What is the expected output? What do you see instead?
You're supposed to see informations about the player (like the name he gave to his character) in the bubble information, in the right corner, on bottom.

What version of the core and database are you using? On what operating system?
Core Rev : acc45bc
DB Rev : latest available
OS : Windows 8.1

Please provide any additional information below.
I know it's not very important, in fact I was wondering why this addon especially was not working !
If you don't know, Total RP 3 is an addon to create a character sheet directly in game.

**Comment by DDuarte** [comment]
@DJScias: could you comment on this?

**Comment by DJScias** [comment]
@DDuarte: Sure I can.

**Mary Sue Protocol**
Basically TRP3 (and a lot of RP chat other addons) communicate through the [Mary Sue Protocol](https://moonshyne.org/msp/) (Addonmsgs->Whispers) and/or xtensionxtooltip2 (Addonmsgs->Channels).

What happens under the hood is every time a user hovers over or is near another user it will send a REQUEST (identified by ?) whisper addon msgs/packets in the form of `?TT?HH?AG?AE?HB?DE?HI?AH?AW?MO?NH?IC?CO` with the MSP prefix. Which is freely translated to 'Do you have MSP info? If yes, please send me it.' MSP from the other client will answer however If the answer to this is no then nothing will happen and MSP will no longer request anything from this client for a while.

**Retail vs Trinity Core**
Now, on retail the following happens:
* The other client will respond to this and use SMSG_CHAT to start sending RESPONSE MSP packets, broken up in several chunks (if > 255 bytes) that are separated with "\1" (0x01 byte). These will be in the form of (in example) `VP93=1VA93=MyRolePlay/6.1.0.401NA95=Sir Tyzen WalkerNHNI93=TyNTRA93=HumanFR95=4FC120=1CU46=Reading.TT136HH93=LakeshireAG94=29AE94=BlueHB93=LordaeronDE97=Tyzen Walker is rough, stern and does not talk much (observant, not shy). Gives an angr`. This is as you can see just a small part of the description as this was chunk 1 (MSP\1).  
* After the above is finished the two MSP clients will communicate every time they come into each other's reach and then, last but not least, the NOT UPDATED packets may arrive. These packets are sent as `!TT1HHAGAEHBDEHIAHAWMONHICCO` and mean that there was no changes and will tell the other MSP client to just 'move on and be done with your business'.

Now, on Trinity Core the following happens:
* Each client with TRP3 will be sending SHITLOADS of REQUEST packets.
* No client ever sends back a RESPONSE (or NOT UPDATED) packet for that matter.

**What is going wrong?**
This means that there is currently a one-way traffic (only REQUEST sent, no RESPONSE) and this effectively stops communication from happening. This is why, if you tried to hackfix TRP3 to work on private servers, you'll notice that it only works client-sided but not if you try to use it on others. Thus there is no proper communication.

**How will we fix this?**
Frankly put.. I've been looking into this issue ever since I committed https://github.com/TrinityCore/TrinityCore/commit/b5aebf40a8a6f4b480ba2e877451db2ac3dac968 a month ago together with @Carbenium , @MitchesD and @DDuarte and it seems that up to this moment we haven't figured out what exactly is causing it.

We've set up breakpoints and checked the contents of the PREFIX packets, the SMSG_CHAT packets, etc.. It all looks normal and nothing seems out of order when we check sniff and parse the packets. I can try keeping you updated but this is a rather low-profile/low-priority issue.

**TL;DR, you talk too much**
I can confirm that this is a thing. TRP3 (through whisper & channel addon msgs) is only sending REQUESTS and no RESPONSES thus no communication can be established.

**Comment by hoxton22** [comment]
Thanks for all the info ! I'd very be pleased to be updated if you find some other interesting things !
Thanks guys :)

**Comment by DDuarte** [comment]
@DJScias: please verify if the issue is fixed

**Comment by DJScias** [comment]
@DDuarte Currently not at a computer, but I will be this weekend and then I'll check and update you through here and IRC. Thanks!

Kind regards,
DJScias

**Comment by Aokromes** [comment]
near 3 months and still no answer, if this bug is still valid, open a new ticket for it.
